autolab/Autolab
# Add hooks to documentation

## Description


Adds a lab hooks section to the documentation

Updates some hooks
- `listOptions`: backwards compatible change that lets default options be correctly hidden. Also adds error handling.
- `autogradeInputFiles`: additionally pass it `assessment` and `submission` as params. A breaking change, but a necessary one and it doesn't seem to be used.

Also makes the following changes to Docs
- Fix syntax highlighting
- Make "code copy" button work locally
- Replace Facebook link with Slack

Adds references to docs in `__defaultAssessment.rb` and fixes errors.

## Motivation and Context



Closes #1926

## How Has This Been Tested?




- View documentation locally, read through lab hooks section and check for any glaring issues.
- Verify that comments in `__defaultAssessment.rb` link (would) link to the right pages.
- Verify new behavior of `listOptions` by viewing documentation and trying to hide default options and/or defining your own options. Also test the error handling for invalid options.

## Types of changes


## Checklist:
